src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java

```diff
@@ -227,7 +239,12 @@ void shouldThrowAlreadyProcessedOn401CP_XPAY() throws JsonProcessingException {
                 .thenReturn(Mono.error(new WebClientResponseException("api error", HttpStatus.UNAUTHORIZED.value(), "Unauthorized", null, null, null)));
 
         /* test */
-        StepVerifier.create(client.requestGeneralAuthorization(authorizationData))
+
```

Comment:
You could make these checks even stronger by adding a `Mockito.verify(client, Mockito.times(0)).requestPostepayAuthorization(authorizationData)`: the assertion you put checks that indeed `PaymentGatewayClient` outputs an empty `Mono`, with the other one you also check that the postepay authorization method is never called 🙂 (see also other instances of this)

Reply:
Solved in [5c3afec](https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/92/commits/5c3afecb6aff7bbdefdbe1defb58fbc5dd79b455)